samples/add-features-with-contingent-values/src/main/java/com/esri/arcgismaps/sample/addfeatureswithcontingentvalues/screens/MainScreen.kt

```diff
@@ -0,0 +1,292 @@
+                        }
+                    ) {
+                        val onBottomSheetStateChanged = { state: SheetState ->
+                            if (!bottomSheetState.isVisible) {
+                                showBottomSheet = false
+                                mapViewModel.clearFeature()
+                            }
+                        }
+                        BottomSheetContents(
+                            mapPoint,
+                            bottomSheetState,
+                            onBottomSheetStateChanged,
+                            uiState,
+                            statusAttributes,
+                            mapViewModel::onStatusAttributeSelect,
+                            protectionAttributes,
+                            mapViewModel::onProtectionAttributeSelect,
+                            sliderControlParameters,
+                            mapViewModel::onBufferSizeSelect,
+                            mapViewModel::validateContingency
+                        )
+                    }
+                }
+            }
+
+            // message dialog view model for displaying error messages
+            mapViewModel.messageDialogVM.apply {
+                if (dialogStatus) {
+                    MessageDialog(
+                        title = messageTitle,
+                        description = messageDescription,
+                        onDismissRequest = ::dismissDialog
+                    )
+                }
+            }
+        }
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun BottomSheetContents(
+    mapPoint: Point?,
+    bottomSheetState: SheetState,
+    onBottomSheetStateChange: (SheetState) -> Unit,
+    uiState: UIState,
+    statusAttributes: List<CodedValue>,
+    onStatusAttributeSelect: (CodedValue) -> Unit,
+    protectionAttributes: List<CodedValue>,
+    onProtectionAttributeSelect: (CodedValue) -> Unit,
+    sliderControlParameters: SliderControlParameters,
+    onBufferSizeSelect: (Int) -> Unit,
+    onApplyButtonClicked: (Point) -> Unit,
+) {
+    val coroutineScope = rememberCoroutineScope()
+    Column(
+        modifier = Modifier.padding(start = 12.dp, end = 12.dp, bottom = 12.dp),
+    ) {
+        Text(
+            text = "Add Feature",
+            style = MaterialTheme.typography.titleMedium,
+            modifier = Modifier.align(Alignment.CenterHorizontally)
+        )
+
+        Spacer(Modifier.size(8.dp))
+        Text("Attributes:")
+
+        // dropdown boxes for selecting feature status/protection
+        AttributeDropdown("Status", uiState.status, statusAttributes, onStatusAttributeSelect)
+        AttributeDropdown("Protection", uiState.protection, protectionAttributes, onProtectionAttributeSelect)
+        Spacer(Modifier.size(8.dp))
+
+        // buffer size displayed and updated in slider
+        var localBufferSize by remember { mutableIntStateOf(uiState.buffer) }
```

Comment:
Naming - what does "local" mean? Could be simply "bufferSize"? Should this state be invalidated when `uiState` changes?
```suggestion
        var bufferSize by remember(key1 = uiState) { mutableIntStateOf(uiState.buffer) }
```

Reply:
Using a key is much better than what I've currently got here!